Sources/SwiftProcessing/Core/Sketch3D/Sketch3DLight.swift

```diff
@@ -0,0 +1,70 @@
+import SceneKit
+import Foundation
+
+
+public extension Sketch {
+    
+    func ambientLight(_ v1: CGFloat, _ v2: CGFloat, _ v3: CGFloat, _ alpha: CGFloat = 1){
+        self.lightNode.light?.type = SCNLight.LightType.ambient
+        self.lightNode.light?.color = CGColor(srgbRed: v1, green: v2, blue: v3, alpha: alpha)
+    }
+    
+    func ambientLight(_ gray: CGFloat, _ alpha: CGFloat = 1){
+        self.lightNode.light?.type = SCNLight.LightType.ambient
+        self.lightNode.light?.color = CGColor(genericGrayGamma2_2Gray: gray, alpha: alpha)
+    }
+    
+    func createLight(_ tag: String, _ lightSCN: SCNLight, _type: String){
+        
+        let newtag = tag
+
+        if var shapeNode = self.currentTransformationNode.getAvailableShape(newtag) {
```

Comment:
either include a TODO in this if statement or change conditional logic to avoid empty conditional